### PR TITLE
chore(atomix): allow backwards compat. proto changes

### LINKED ISSUE #5223: Allow backwards compatible changes in network protocol
**Description**

Issue #5038 made it possible to make changes to the objects that are stored to disk in a backwards compatible way. We also want to do the same for objects that are serialized and used in the network. For a longer discussion on this topic see #5104 and #5038.

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Enable us to do backwards compatible changes to the protocol. PR https://github.com/zeebe-io/zeebe/pull/5244 prepared 0.24 for these changes so this PR shouldn't break the rolling upgrade path from 0.24 to 0.25. The RollingUpdateTest is using 0.24.4 to test the rolling upgrade but 0.24.4 wasn't released yet so the test won't pass. When 0.24.4 is released, this PR can be marked ready for review.

## Related issues

closes #5223

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [x] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -129,7 +129,7 @@
     <extension.version.os-maven-plugin>1.6.2</extension.version.os-maven-plugin>
 
     <!-- version against which backwards compatibility is checked -->
-    <backwards.compat.version>0.24.0</backwards.compat.version>
+    <backwards.compat.version>0.24.4</backwards.compat.version>
```
Comment (deepthidevaki) [inline parent/pom.xml:132]: RollingUpdateTest is currently disabled. We should enable them https://github.com/zeebe-io/zeebe/blob/0694fe7a0316f47009b7d1d24c9cb37d2f120223/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java#L278

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
LGTM :+1: RollingUpdateTest is still disabled. Since 0.24.4 is not released, we can wait until it is released. Please re-enable RollingUpdateTest before merging.
PS:- PR is not approved because we decided to wait until 0.24.4 is released.

**Comment by npepinpe** [comment]
Artifacts for 0.24.4 should be available at this point :+1:

**Comment by npepinpe** [comment]
As 0.24.4 is available (though not publicly released yet - tomorrow), can we merge this?

**Review by deepthidevaki — APPROVED** [review]
:100:

**Pull request merged**